app/models/accounts/duty.rb

```diff
@@ -4,9 +4,22 @@ class Duty < ApplicationRecord
   # relations
   belongs_to :employee
   belongs_to :unit
+  has_many :positions, dependent: :destroy
+  has_many :administrative_functions, through: :positions
 
   # validations
-  validates :temporary, presence: true
+  validates :temporary, inclusion: { in: [true, false] }
   validates :start_date, presence: true
-  validates :unit_id, uniqueness: { scope: %i[employee] }
+  validates :unit_id, uniqueness: { scope: %i[employee start_date] }
+  validates_with DutyValidator
+
+  # scopes
+  scope :temporary, -> { where(temporary: true) }
+  scope :tenure, -> { where(temporary: false) }
+  scope :active, -> { where('end_date > ? OR end_date IS NULL', Time.zone.today) }
```

Comment:
Kendime ve herkese not, Rails'te bu gibi SQL sorguları yazmak yerine Arel kullanmalıyız. Bunun yerine şöyle olmalıydı:

```ruby
duties = Duty.arel_table
scope :active, -> { where(duties[:end_date].gt(Time.zone.today).or(duties[:end_date].eq(nil))) }
```

Neden peki? Bu sorgunun çalıştığını ve doğru çalıştığını biliyorum, ama Postgres için. MySQL, MsSQL ve diğerlerinde denemedim. Veritabanları arası SQL cümleciği yazma standartları değişiyor. Mesela `LIKE` buna iyi bir örnek:

Postgres'te `LIKE` ifadesi case-sensitive. MySQL'de ise çoğunlukla case-sensitive **değil**. Postgres'te case-sensitive olmayan bir `LIKE` yazmak için `ILIKE` kullanmak gerekiyor. Bu ifade ise MySQL'de tamamen tanımsız. Yani yarın bir gün database adaptörünü MySQL olarak değiştirirsek `ILIKE` yazdığımız yerlerin tümü patlar.

Arel tam burada devreye giriyor. LIKE yerine Arel'in `matches("text:%")` ifadesini kullansak kodumuz farklı database'ler için kırılgan hale gelmemiş olacaktı.